pkg/db-repo/repository.go

```diff
@@ -89,7 +91,14 @@ func (r *repository) Create(ctx context.Context, value ModelBased) error {
 
 	err := r.orm.Create(value).Error
 
-	// TODO: Check for duplicate error and return a local error type instead which applications can handle themselves
+	isDuplicate := db.IsDuplicateEntryError(err)
+	if isDuplicate {
```

Comment:
i would it write as `if db.IsDuplicateEntryError(err) {`  
if the condition statement gets more complex it makes sense to have it on a separate line but for simple ones i find it more readable to have it inline..